components/mixins/FixedTop.js

```diff
@@ -8,36 +9,31 @@ export default {
       defaultDebounceTime: 200,
       prevScrollY: 0,
       fixedTop: null,
-      currScrollY: 0,
       isMobileMenuShown: false
     }
   },
   components: {},
   props: {},
-  computed: {
-    isScrollingDown () {
-      return this.currScrollY > this.prevScrollY
+  computed: {},
+  methods: {
+    isScrollingUp (currScroll) {
+      return currScroll < this.prevScrollY
     },
-
-    isScrollingUp () {
-      return this.currScrollY < this.prevScrollY
+    isEligibleForScroll (currScroll) {
+      return currScroll > this.initializeHideHeight
     },
+    fixedTopFall: _.debounce(function (e) {
+      const curr = window.scrollY
 
-    isEligibleForScroll () {
-      return this.currScrollY > this.initializeHideHeight
-    }
-  },
-  methods: {
-    fixedTopFall: _.debounce(function () {
-      this.currScrollY = window.scrollY
-
-      if (!this.isMobileMenuShown && this.isEligibleForScroll && this.isScrollingDown) {
-        this.hideFixedTop()
-      } else if (this.isScrollingUp) {
-        this.showFixedTop()
+      if (!this.isMobileMenuShown && this.isEligibleForScroll(curr)) {
+        if (this.isScrollingUp(curr)) {
+          this.showFixedTop()
+        } else {
```

Comment:
It's wrong it should not scroll down when this.currScroll = this.prevScroll also i do not like the double if. Please change it back